.circleci/config.yml

```diff
@@ -1,28 +1,67 @@
 version: 2
+
 jobs:
-  build:
+  checkout_code_and_dependencies:
+    machine: true
+    working_directory: ~/repo
+    steps:
+      - checkout
+      - run:
+          name: Download project code and dependency jars
+          command: ./gradlew resolveAllDependencies
```

Comment:
what's the purpose of this step?

Reply:
I think if you do that, then you should:
1. clone the repo
2. resolve dependencies
3. cache the repo
4. cache the gradle dependencies

Then in subsequent steps:
1. restore repo and gradle caches
2. build

Reply:
I dont think its a good idea to cache test results but better upload it from the task itself.

Reply:
Good point. Makes sense to upload at task level itself instead of cache, merge and upload.